x/lightclient/keeper/rollback.go

```diff
@@ -2,24 +2,117 @@ package keeper
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	"github.com/dymensionxyz/dymension/v3/x/lightclient/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
+
+	ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
 )
 
-func (hook rollappHook) OnHardFork(sdk.Context, string, uint64) error {
-	/*
-		// freeze IBC client state
-		clientState, ok := k.ibcClientKeeper.GetClientState(ctx, clientId)
-		if !ok {
-			return errorsmod.Wrapf(types.ErrInvalidClientState, "client state for clientID %s not found", clientId)
-		}
+func (hook rollappHook) OnHardFork(ctx sdk.Context, rollappId string, height uint64) error {
+	hook.k.RollbackCanonicalClient(ctx, rollappId, height)
+	return nil
+}
+
+func (k Keeper) RollbackCanonicalClient(ctx sdk.Context, rollappId string, height uint64) {
```

Comment:
not clear if `height` is target height or fraud height. I suggest renaming it accordingly.